### PR TITLE
ticket 5

### PR DESCRIPTION
**Pull request by Masonward99**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -68,6 +69,44 @@ describe('Get/api/reviews/:review_id', () => {
...
+                    expect(result).toHaveProperty('owner');
+                    expect(result).toHaveProperty("title");
+                    expect(result).toHaveProperty("review_id");
+                    expect(result).toHaveProperty("category");
+                    expect(result).toHaveProperty("review_img_url");
+                    expect(result).toHaveProperty("votes");
+                    expect(result).toHaveProperty("created_at");
+                    expect(result).toHaveProperty("comment_count");
+                    expect(result).toHaveProperty("created_at");
+                })
```
Comment (EddTheDuck) [inline __tests__/app.test.js:89]: Good that you have checked for presence of each property but you should be checking for data type also

File: __tests__/app.test.js [hunk 2]
```diff
@@ -68,6 +69,44 @@ describe('Get/api/reviews/:review_id', () => {
...
+    it('does not have review_body property', () => {
+        return request(app)
+            .get('/api/reviews')
+            .expect(200)
+            .then((results) => {
+                let data = results.body.reviews;
+                data.forEach(result => {
+                    expect(result).not.toHaveProperty('review_body')
+                })
+            })
```
Comment (EddTheDuck) [inline __tests__/app.test.js:101]: this might as well be an extra assertion in the previous test

File: controllers/reviews.controllers.js [hunk 3]
```diff
@@ -1,8 +1,14 @@
...
     findReviewById(id)
         .then(review => res.status(200).send({ review }))
         .catch(next);
+}
+
+exports.getReview = (req, res, next) => {
+    findReview()
+        .then(reviews => res.status(200).send({ reviews }))
+        .catch(next)
 }
```
Comment (EddTheDuck) [inline controllers/reviews.controllers.js:14]: For naming variables/functions try be more specific - getReviews is better than getReview

File: models/reviews.models.js [hunk 4]
```diff
@@ -13,4 +15,23 @@ exports.findReviewById = (id) => {
...
+            for (let i = 0; i < formatedCount.length; i++){
+                connection.query(
+                  `UPDATE reviews SET comment_count = $1 WHERE review_id = $2 RETURNING *;`,
+                  formatedCount[i]
+                );
+            }
+        })
+        .then(() => connection.query(`SELECT review_id, title, category, designer, owner, review_img_url, created_at, votes, comment_count FROM reviews ORDER BY created_at DESC`))
+    .then(res=> res.rows)
+}
```
Comment (EddTheDuck) [inline models/reviews.models.js:37]: You shouldn;t be altering the database tables at all. I believe this is only passing becausethe db is being reseeded before each test - can't add the same  column everytime.

**Review by EddTheDuck — CHANGES_REQUESTED** [review]
Some changes needed here - see comments and resubmit a PR when complete.

File: __tests__/app.test.js [hunk 5]
```diff
@@ -68,6 +73,46 @@ describe('Get/api/reviews/:review_id', () => {
             .then(res => expect(res.body.msg).toBe('bad request'))
     })
 })  
+describe('GET/api/reviews', () => {
+    it('should return an array of review objects', () => {
+        return request(app)
+          .get("/api/reviews")
+          .expect(200)
+            .then((results) => {
+                let data = results.body.reviews;
```
Comment (yalicee) [inline __tests__/app.test.js:82]: check the length of the array if you are going to use a forEach

**Review by yalicee — APPROVED** [review]
Make sure to make changes from coments before merging :)

**Pull request merged**